src/nvim/edit.c

```diff
@@ -3752,7 +3735,7 @@ static bool ins_bs(int c, int mode, int *inserted_space_p)
   }
 
   // Delete newline!
-  if (curwin->w_cursor.col == 0) {
+  if (curwin->w_cursor.col <= revins_on) {
```

Comment:
It seems that this change can cause Nvim to crash. Please remove this change from this PR.